docs/details/tests.md

```diff
@@ -32,3 +32,113 @@ The command `memtest` uses this idea to find memory leaks in pure functions. How
 
 If the function fails, the command will ignore it and still test the function for leaks. Side-effects shouldn't happen unconditionally, even if the function fails. If you want to find unexpected failures, use `test` command instead.
+
+## Constant value for arguments
+
+The `deal.cases` constructor accepts `kwargs` argument where you can specify constant values for the function arguments. For example:
+
+```python
+@deal.raises(ZeroDivisionError)
+def div(a: int, b: int) -> float:
+    assert a == 1
+    return a / b
+
+# Every test case calls `div` function with `a=1`.
+# So, random values are generated only for `b`.
+@deal.cases(div, kwargs=dict(a=1))
+def test_div(case):
+    case()
+```
+
+## Custom strategies
+
+Under the hood, `deal.cases` uses [hypothesis](https://hypothesis.readthedocs.io/en/latest/index.html) testing framework to generate test cases. The trick is that `kwargs` argument of `deal.cases` can contain hypothesis strategies:
+
+```python
+import hypothesis.strategies as st
+
+@deal.raises(ZeroDivisionError)
+def div(a: int, b: int) -> float:
+    assert a >= 10
+    return a / b
+
+cases = deal.cases(
+    func=div,
+    kwargs=dict(
+        a=st.integers(min_value=10),
+    ),
+)
+for case in cases:
+    case()
+```
+
+## Reproducible failures
+
+Argument `seed` of `deal.cases` is a random seed. That means, for the same seed value you always get the same test cases. It is a must-have thing for CI. There are a few important things:
+
++ If the seed is different for diferrent pipelines, it will run a bit different test cases every time which increases your chances to find a tricky corner case.
++ If the seed is the same when you re-run the same job, it will help you to identify flaky tests. In other words, if a CI job fails, you re-run it, and it passes, it was a flaky test which happens because of tricky side-effects (database connection failed, another test changed a state etc.) and it will be hard to reproduce. However, if re-run fails with the same error, most probably it is a failure that you can easily reproduce and debug locally, knowing the seed.
++ The seed should be shown in the CI job to make it possible to use it locally to reproduce a failure. It is either printed in the job output or is something known like the pipeline run number.
+
+There is an example for GitLab CI:
+
+```python
+import sys
+import deal
+
+seed = None
+if os.environ.get('CI'):
+    seed = int(os.environ['CI_PIPELINE_ID'])
+
+@deal.cases(div, seed=seed)
+def test_div(case):
+    case()
+```
+
+## Fuzzing
+
+[Fuzzer](https://en.wikipedia.org/wiki/Fuzzing) is when an external tool or library (fuzzer) generates a bunch of random data in hope to break your program. That means, fuzzing requires a lot of resources and is performance-critical. This is why most of the fuzzers are written on C. However, there are few Python wrappers for existing fuzzers, with awful interface but working:
```

Comment:
> with awful interface but working

Maybe we can make this part a bit more friendly?

Reply:
You're right, let's be more friendly :) The reason why I say "awful" is that I still can't figure out why to pass `sys.argv` into atheris. I tried to search issues, readme, just to experiment with it, try to pass `--help` inside. No effect. The only thing I managed to do through atheris API is just to run the test function with defaults.

Reply:
> This is a good point, and something I've thought of before. They are actually separate for a reason though: `atheris.Setup()` might modify the input argument list, by removing arguments handled by libFuzzer. This allows other argument parsing libraries to subsequently be called and not think there are a lot of excess arguments.

> We use this feature inside of Google. We call `atheris.Setup()` to remove libFuzzer arguments, then pass `sys.argv` to our argument-parsing code to initialize Google infrastructure, then finally call `atheris.Fuzz()`.

_Originally posted by @IanPudney in https://github.com/google/atheris/issues/8#issuecomment-746434515_

Reply:
@Zac-HD is `hypofuzz` that project you told me about in the summer? This looks really interesting! 

I would love to test this out whenever you need my help.

Reply:
That's the one!  I'll be in touch in the new year 😁